File: backend-api/src/main/java/fr/pmeflow/backend/dossier/DossierService.java
/*******************************************************************************
 *    
 *                           APPLICATION Pmeflow
 *                          =======================
 *   
 *   @AUTHOR Pmeflow.legal
 *
 ******************************************************************************/
package fr.pmeflow.backend.dossier;

import java.util.List;

import javax.ws.rs.Consumes;
import javax.ws.rs.GET;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.PathParam;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;

import fr.pmeflow.backend.dossier.model.Dossier;
import fr.pmeflow.backend.io.model.Document;
import fr.pmeflow.commons.model.enumeration.Langue;
import fr.lixbox.common.exceptions.BusinessException;
import fr.lixbox.service.common.MicroService;

/**
 * Cette interface represente les fonctions disponibles
 * dans le service Dossier.
 * 
 * @author ludovic.terral
 */
@Path(DossierService.SERVICE_URI)
@Produces({ MediaType.APPLICATION_JSON+"; charset=utf-8" })
@Consumes({ MediaType.APPLICATION_JSON+"; charset=utf-8" })
public interface DossierService  extends MicroService
{  
    // ----------- Attribut(s) -----------   
    static final long serialVersionUID = 1306745024315L;
    static final String SERVICE_NAME = "pmeflow-service-api-dossier";
    static final String SERVICE_VERSION = "1.0";
    static final String SERVICE_URI = SERVICE_VERSION+"/dossier";
    static final String FULL_SERVICE_URI = "/backend/api/"+SERVICE_URI;
    static final String SERVICE_CODE = "DOSSERV";
    
    

    // ----------- Methodes -----------
    @GET @Path("/{langue}/verif/create") boolean verifDossierCreate(@PathParam("langue") Langue langue) throws BusinessException;
    @GET @Path("/{langue}/all") List<Dossier> getDossiers(@PathParam("langue") Langue langue) throws BusinessException;
    @GET @Path("/{langue}/actif") List<Dossier> getDossiersActifs(@PathParam("langue") Langue langue) throws BusinessException;
    @GET @Path("/{langue}/{id}") Dossier getDossierById(@PathParam("langue") Langue langue, @PathParam("id") String id) throws BusinessException;
    @POST @Path("/{langue}") Dossier mergeDossier(@PathParam("langue") Langue langue, Dossier dossier) throws BusinessException;
    @POST @Path("/{langue}/{id}/document") boolean addDocumentToDossier(@PathParam("langue") Langue langue, @PathParam("id") String id, Document document) throws BusinessException;
}

File: private-ui/src/api/DossierService.js
import axios from 'axios';

export default class DossierService {
    API_URL = process.env.VUE_APP_DOSSIER_SERVICE_URI;
    token = "";
    
    constructor(url, token) {
        this.API_URL = url;
        this.token = token;
    }


    verifDossierCreate() {
        const url = `${this.API_URL}/FR_FR/verif/create`;
        return axios
            .get(url, {
                headers: {
                    "content-type": "application/json",
                    authorization: "Bearer " + this.token,
                }}
            );
    }


    getDossiers() {
        const url = `${this.API_URL}/FR_FR/all`;
        return axios
            .get(url, {
                headers: {
                    "content-type": "application/json",
                    authorization: "Bearer " + this.token,
                }}
            );
    }


    
    getDossiersActifs() {
        const url = `${this.API_URL}/FR_FR/actif`;
        return axios
            .get(url, {
                headers: {
                    "content-type": "application/json",
                    authorization: "Bearer " + this.token,
                }}
            );
    }



    getDossierById(id) {
        const url = `${this.API_URL}/FR_FR/${id}`;
        return axios
            .get(url, {
                headers: {
                    "content-type": "application/json",
                    authorization: "Bearer " + this.token,
                }}
            );
    }

    
    mergeDossier(dossier) {
        const url = `${this.API_URL}/FR_FR`;
        return axios
            .post(url, dossier,
                {
                headers: {
                  "content-type": "application/json",
                  authorization: "Bearer " + this.token,
                },
              }
            );
    }

    addDocumentToDossier(dossierId, document) {
        const url = `${this.API_URL}/FR_FR/${dossierId}/document`;
        return axios
            .post(url, document,
                {
                headers: {
                    "content-type": "application/json",
                    authorization: "Bearer " + this.token,
                },
                }
            );
    }
}